Sources/XCTest/Public/XCTestCase.swift

```diff
@@ -194,26 +195,24 @@ open class XCTestCase: XCTest {
     /// Teardown method called after the invocation of every test method in the
     /// class.
     open class func tearDown() {}
-
-    private var teardownBlocks: [() -> Void] = []
-    private var teardownBlocksDequeued: Bool = false
-    private let teardownBlocksQueue: DispatchQueue = DispatchQueue(label: "org.swift.XCTest.XCTestCase.teardownBlocks")
-
+    
     /// Registers a block of teardown code to be run after the current test
     /// method ends.
     open func addTeardownBlock(_ block: @escaping () -> Void) {
-        teardownBlocksQueue.sync {
-            precondition(!self.teardownBlocksDequeued, "API violation -- attempting to add a teardown block after teardown blocks have been dequeued")
-            self.teardownBlocks.append(block)
-        }
+        teardownBlocksState.append(block)
+    }
+    
+    /// Registers a block of teardown code to be run after the current test
+    /// method ends.
+    @available(macOS 12.0, *)
+    open func addTeardownBlock(_ block: @Sendable @escaping () async throws -> Void) {
```

Comment:
I just remembered, the Darwin/Xcode copy of this API is `public` and not `open` since it's in a Swift overlay. I think it may be safer to match that here, and make this `public`. It'll be slightly more conservative which is probably safer for now anyway
```suggestion
    public func addTeardownBlock(_ block: @Sendable @escaping () async throws -> Void) {
```